Scripts/General/ModMercs.lua

```diff
@@ -0,0 +1,732 @@
+    local UpgradeDmg1Level      = Merc_GetUpgradeLevel(Merc, EMercUpgradeType.MeleeDamage) + 1
+    --local UpgradeDmg1Level      = GetUpgradeLevel(Merc, EMercUpgradeType.MageRes)
+
+    -- Monster
+    local mon                   = SummonMonster(Merc.MonsterID, Party.X, Party.Y, Party.Z, false)
+    
+    -- User values
+    mon.FullHitPoints           = Merc.FullHP[UpgradeHPLevel]
+    mon.ArmorClass              = Merc.AC[UpgradeHPLevel]
+    mon.Level                   = Merc.Level[UpgradeHPLevel]
+
+    -- Default Variables
+    mon.NPC_ID                  = QuestNPC
+    mon.Group                   = 35
+    mon.Hostile                 = false
+    mon.Ally                    = 9999
+    mon.Summoner                = 28
+    mon.HostileType             = 0
+    mon.Experience              = 0
+    mon.ShowOnMap               = true
+    mon.HP                      = mon.FullHitPoints
+    mon.NoFlee                  = true
+    
+    -- Damage
+    local meleeDmg              = Merc_ParseDamageString(Merc.Attack1[UpgradeDmg1Level])
+    mon.Attack1.DamageAdd       = meleeDmg.DamageAdd
+    mon.Attack1.DamageDiceCount = meleeDmg.DamageDiceCount
+    mon.Attack1.DamageDiceSides = meleeDmg.DamageDiceSides
+
+    ExitScreen()
+end
+
+function Merc_Dismiss(Merc, t)
+
+    evt.Add("NPCs", QuestNPC)
+
+    local MercSaveData      = Merc_GetSaveDataByID(Merc.NPC_ID)
+    MercSaveData.Released   = false
+    MercSaveData.ReleaseMap = ""
+    
+    for _, mon in Map.Monsters do
+        if mon.NPC_ID == QuestNPC then
+            RemoveMonster(mon)
+        end
+    end
+
+    -- Reset
+    Timer(
+        function()
+            local UpgradeUseLevel   = Merc_GetUpgradeLevel(Merc, EMercUpgradeType.Charges) or 1
+            local MercSaveData      = Merc_GetSaveDataByID(Merc.NPC_ID)
+            MercSaveData.FightsLeft = Merc.FightsMax[UpgradeUseLevel+1] or 1
+        end, 
+        const.Day, const.Hour, true)
+
+    ExitScreen()
+end
+
+function Merc_IsSpecial(Merc, t)
+    
+    if Merc.Ability == nil              then return false end
+    if type(Merc.Ability) ~= "string"   then return false end
+    if string.len(Merc.Ability) == 0    then return false end
+    return true
+end
+
+function Merc_IsSpecialTag(Merc, tag)
+
+    if not Merc_IsSpecial(Merc, nil) then return false end
+
+    for word in string.gmatch(tag, "([^,]+)") do
+        if word == tag then
+            return true
+        end
+    end
+
+    return false
+end
+
+function Merc_ShowMessageAboutAbility(Merc, t) 
+
+    if Merc_IsSpecial(Merc, t) then
+        Message(Merc.Credentials.TextSpecial)
+    end
+end
+
+function Merc_ConsumeCharge(Merc)
+
+    local MercSaveData = Merc_GetSaveDataByID(Merc.NPC_ID)
+    MercSaveData.FightsLeft = math.max(0, MercSaveData.FightsLeft - 1)
+end
+
+function Merc_ParseDamageString(damageString)
+
+    local count, sides, add = damageString:match("(%d+)D(%d+)%+?(%d*)")
+
+    count   = tonumber(count)
```

Comment:
if damageString can be corrupted only because someone put garbage data in txt tables then I think its ok to not check the types and handle exception as you want the lua runtime to fail. However if you want to have some human friendly exception here then you should handle those exceptions.